### PR TITLE
address issue #221: drop regressors that are collinear with the FE depending on tolerance

### PR DESCRIPTION
**Pull request by moritzdrechselgrau**

### CODE REVIEW

**Comment by matthieugomez** [comment]
Thanks! Could you add a test?

File: src/utils/basecol.jl [hunk 1]
```diff
@@ -76,3 +76,24 @@ end
 function getcols(X::AbstractMatrix,  basecolX::AbstractVector)
     sum(basecolX) == size(X, 2) ? X : X[:, basecolX]
 end
+
+
+##############################################################################
+# Check whether a variable is likely collinear with the fixed effects
+##############################################################################
+@inline function sumofsquares(@nospecialize(xs::AbstractVector...))
+    [x' * x for x in xs]
```
Comment (matthieugomez) [inline src/utils/basecol.jl]: Is defining these one liner functions really necessary? Just using `[sum(abs2, x) for x in eachcol(Xall)]` in fit.jl seems more transparent

Comment (moritzdrechselgrau) [reply src/utils/basecol.jl]: I've added the example in issue 221 as a test.

**Comment by moritzdrechselgrau** [comment]
I've added a test using the example in issue #221 and got rid of the one-liner functions.

**Comment by matthieugomez** [comment]
Awesome! thanks a lot for your contribution.

**Comment by moritzdrechselgrau** [comment]
Sure! Thanks for the package!

**Pull request merged**